src/Symfony/Component/Security/Core/Authentication/RememberMe/CacheTokenVerifier.php

```diff
@@ -45,12 +45,12 @@ public function verifyToken(PersistentTokenInterface $token, string $tokenValue)
         }
 
         $cacheKey = $this->getCacheKey($token);
-        if (!$this->cache->hasItem($cacheKey)) {
-            return false;
-        }
-
         $item = $this->cache->getItem($cacheKey);
+
         $outdatedToken = $item->get();
+        if (!\is_string($outdatedToken)) {
```

Comment:
this should be moved one line above and be like this:
```php
if (!$item->isHit()) {
    return false;
}
```

Reply:
@nicolas-grekas why is it better? `is_string` narrows down the type of `$outdatedToken` to a string, while `!$item->isHit()` would leave it as `mixed`.

Reply:
Just because it makes more sense when reading the code. I don't think the responsibility of these lines should be to guard against the case you describe.